test/e2e/specs/editor/editor__post-basic-flow.ts

```diff
@@ -160,16 +162,34 @@ describe( DataHelper.createSuiteTitle( 'Editor: Basic Post Flow' ), function ()
 		skipItIf( envVariables.VIEWPORT_NAME === 'mobile' )( 'Save draft', async function () {
 			await editorPage.saveDraft();
 		} );
+
+		it( 'Publish post', async function () {
+			publishedURL = await editorPage.publish();
+		} );
 	} );
 
-	describe( 'Publish', function () {
-		it( 'Publish and visit post', async function () {
-			const publishedURL: URL = await editorPage.publish( { visit: true } );
-			expect( publishedURL.href ).toStrictEqual( page.url() );
+	// Skip test on Private site, because posts are not visible to non-logged out users.
+	skipDescribeIf( envVariables.ATOMIC_VARIATION === 'private' )( 'View post', function () {
+		let newPage: Page;
+
+		beforeAll( async function () {
+			newPage = await browser.newPage();
+		} );
+
+		it( 'View published post', async function () {
+			// Check for `blog` and `post` query params, used for stats tracking.
+			const trackingPixelLoaded = newPage.waitForResponse(
+				/pixel\.wp\.com\/g\.gif.*blog=[\d]+.*&post=[\d]+/
+			);
+			await newPage.goto( publishedURL.href );
+			const response = await trackingPixelLoaded;
+			expect( response.status() ).toBe( 200 );
```

Comment:
The issue with starting the promise in a step after the actual navigation to the post, is that `waitForResponse` doesn't have visibility into requests that completed prior to calling this method. As [upstream documentation](https://playwright.dev/docs/network#variations) suggests, the `waitForResponse` has to wrap around the action triggering network requests.

Case in point, this is what happens if the `waitForResponse` call is moved into a new step after the `Visit Published Post` step. This is run on my local machine:

```
 page.waitForResponse: Timeout 10000ms exceeded while waiting for event "response"
    =========================== logs ===========================
    waiting for response /pixel.wp.com\/g.gif.*blog=212495662+.*&post=[\d]+/
    ============================================================
```

Reply:
I worked around the issue and updated the PR. 

The idea is to wrap the `waitForResponse` in a try/catch, assign the response to a holding value, and then check whether it's defined in the next step. This way if the tracking pixel ever fails to load, it will not fail at the "Visit Published Post" step but rather the "Jetpack Stats tracking pixel is loaded" step.

Reply:
> As [upstream documentation](https://playwright.dev/docs/network#variations) suggests, the waitForResponse has to wrap around the action triggering network requests.

> Case in point, this is what happens if the waitForResponse call is moved into a new step after the Visit Published Post step.

Bah, I should have made myself clearer, sorry! I just meant passing the promise onto the next step instead of awaiting it in that one. I definitely think we need to keep it wrapped around.

So, something like...

```typescript
let trackingPixelLoadedPromise: Promise< Response >;
it( 'View published post', async function () {
  // Check for `blog` and `post` query params, used for stats tracking.
  trackingPixelLoadedPromise = newPage.waitForResponse(
  new RegExp(
	  `pixel.wp.com/g.gif.*blog=${ testAccount.credentials.testSites?.primary.id }+.*&post=[\\d]+`
  )
  );
  await newPage.goto( publishedURL.href );
 
  expect( publishedURL.href ).toStrictEqual( newPage.url() );
  } );
  
it( 'Jetpack Stats tracking pixel is loaded', async function () {
  const response = await trackingPixelLoadedPromise;
  expect( response ).toBeDefined();
  expect( response.status() ).toBe( 200 );
} );
```

But, what you have totally works too and is fine! 👍

Reply:
> So, something like...
> 
> 
> 
> ```typescript
> 
> let trackingPixelLoadedPromise: Promise< Response >;
> 
> it( 'View published post', async function () {
> 
>   // Check for `blog` and `post` query params, used for stats tracking.
> 
>   trackingPixelLoadedPromise = newPage.waitForResponse(
> 
>   new RegExp(
> 
> 	  `pixel.wp.com/g.gif.*blog=${ testAccount.credentials.testSites?.primary.id }+.*&post=[\\d]+`
> 
>   )
> 
>   );
> 
>   await newPage.goto( publishedURL.href );
> 
>  
> 
>   expect( publishedURL.href ).toStrictEqual( newPage.url() );
> 
>   } );
> 
>   
> 
> it( 'Jetpack Stats tracking pixel is loaded', async function () {
> 
>   const response = await trackingPixelLoadedPromise;
> 
>   expect( response ).toBeDefined();
> 
>   expect( response.status() ).toBe( 200 );
> 
> } );
> 
> ```

I actually did what you suggested as the first step, and it was what produced the error log that I pasted. 

It does align with previous experience re:waitForResponse behavior. The request must fire after the waitForResponse call starts, and since the request starts soon after the page is loaded the separated call ends up timing out.